sdk/core/core/src/az_json_string.c

```diff
@@ -74,49 +74,6 @@ AZ_NODISCARD AZ_INLINE az_result az_json_esc_decode(uint8_t c, uint8_t* out)
   return AZ_OK;
 }
 
-/**
- * Encodes the given character into a JSON escape sequence. The function returns an empty span if
- * the given character doesn't require to be escaped.
- */
-AZ_NODISCARD az_span _az_json_esc_encode(uint8_t c)
```

Comment:
I am surprised we are able to get rid of this logic. I would have expected this to be used when trying to add JSON string literals using the builder. We don't want folks to be able to write invalid JSON.

For example, having un-escaped new lines within a JSON string value is invalid.
```json
["hello\nWorld"]
```